enhancements/update/targeted-update-edge-blocking.md

```diff
@@ -0,0 +1,400 @@
+    The query should return a 0 if the update should be allowed and a 1 if the update should be blocked.
+  * FIXME: [`platform` or other properties that are easier to evaluate than PromQL?](#non-promql-filters)
+
+[The schema version][graph-data-schema-version] would also be bumped to 1.1.0, because this is a backwards-compatible change.
+Consumers who only understand graph-data schema 1.0.0 would ignore the `clusters` property and block the edge for all clusters.
+The alternative of failing open is discussed [here](#failing-open).
+
+### Enhanced Cincinnati JSON representation
+
+[The Cincinnati graph API][cincinnati-api] will be extended with a new top-level `conditionalEdges` property, with an array of conditional edge [objects][json-object] using the following schema:
+
+* `from` (required, [string][json-string]), with the `version` of the starting node.
+* `to` (required, [string][json-string]), with the `version` of the ending node.
+* `qualifiers` (optional, [array][json-array], with qualifications around the recommendation.
+  Each entry is an [object][json-object] with the following schema:
+  * `url` (optional, [string][json-string]), with a URI documenting the issue, as described in [the blocked-edges section](#enhanced-graph-data-schema-for-blocking-edges).
+  * FIXME: [also reason/message?](#metadata-to-include-when-blocking-a-conditional-request)
+  * `clusters` (optional, [object][json-object]).
+    If any `clusters` property matches a given cluster, the edge is not recommended for that cluster, as described in [the blocked-edges section](#enhanced-graph-data-schema-for-blocking-edges).
+    If `clusters` is unset, the edge is blocked for all clusters.
+    * `promql` (optional, [object][json-object]), defining the subset of affected clusters, as described in [the blocked-edges section](#enhanced-graph-data-schema-for-blocking-edges).
+    * FIXME: [`platform` or other properties that are easier to evaluate than PromQL?](#non-promql-filters)
+
+### Enhanced ClusterVersion representation
+
+[The ClusterVersion `status`][api-cluster-version-status] will be extended with a new `conditionalUpdates` property:
+
+```go
+// conditionalUpdates contains the list of updates that are
+// conditionally recommended for this cluster. Consumers interested
+// in the set of updates that are actually recommended for this
+// cluster should use availableUpdates. This list may be empty if no
+// updates are recommended, if the update service is unavailable, or
+// if an empty or invalid channel has been specified.
+// +optional
+conditionalUpdates []ConditionalUpdate `json:"conditionalUpdates,omitempty"`
+```
+
+The `availableUpdates` documentation will be adjusted to read:
+
+```go
+// availableUpdates contains the subset of conditionalUpdates that
+// apply to this cluster.  Updates which appear in conditionalUpdates
+// but not in availableUpdates may expose this cluster to known
+// issues.
```

Comment:
I do not like the current text as it is not user friendly. It should be 
```
availableUpdates contains the updates or versions which is recommended by the OpenShift Update Service (OSUS) for this cluster.
```

Reply:
We can add that there may be more versions available in general but not recommended for this cluster for cluster specific configurations and for more details check the the other property i.e. conditionalUpdates

Reply:
> ...OpenShift Update Service (OSUS) for this cluster...

OSUS is our sub-product.  The cluster may be configured to point a a different `upstream` update service.  [The `upstream` godocs][1] use "update server", which I'm fine softening to "update service" (clients should not care if the service is backed by multiple servers or a single server).  But I'd rather keep our implementation-specific name out of these implementation-agnostic docs.  "subset" is wrong, though; I'll reroll to drop that.

[1]: https://github.com/openshift/api/blob/6318df75be26bf34a80ecd098ccd5a5454e3aa62/config/v1/types_cluster_version.go#L56